extra_data/validation.py

```diff
@@ -193,24 +219,23 @@ def progress(self, line):
     def check_files(self):
         self.file_accesses = []
 
-        for i, filename in enumerate(self.filenames):
-            self.progress(f"{i}/{len(self.filenames)} files "
-                          f"({len(self.problems)} problems): {filename}")
-            path = osp.join(self.run_dir, filename)
-            try:
-                fa = FileAccess(path)
-            except Exception as e:
-                self.problems.append(
-                    dict(msg="Could not open file", file=path, error=e)
-                )
-            else:
-                self.file_accesses.append(fa)
-                fv = FileValidator(fa)
-                self.problems.extend(fv.run_checks())
-                fa.close()
+        def initializer():
+            # prevent child processes from receiving KeyboardInterrupt
+            signal(SIGINT, SIG_IGN)
+
+        filepaths = [osp.join(self.run_dir, fn) for fn in self.filenames]
+        nfiles = len(self.filenames)
+        with Pool(initializer=initializer) as pool:
+            for fa, problems in pool.imap_unordered(_check_file, filepaths):
+                self.problems.extend(problems)
+                if fa is not None:
+                    self.file_accesses.append(fa)
+                    fa.close()
 
-        self.progress("{0}/{0} files".format(len(self.filenames)))
+                    self.progress(
+                        len(self.file_accesses), nfiles, len(self.problems))
```

Comment:
This should be outside the `if` block,  and with a separate counter rather than using `len(self.file_accesses)`.  A file that can't be opened still gets you closer to finishing validation.